lib/SILOptimizer/Transforms/DeadCodeElimination.cpp

```diff
@@ -40,6 +40,11 @@ namespace {
 // FIXME: Reconcile the similarities between this and
 //        isInstructionTriviallyDead.
 static bool seemsUseful(SILInstruction *I) {
+  // We have a reverse dependency for end_access. If there are no uses of a
+  // begin_access, both will be removed.
```

Comment:
nitpick: Can you change this comment to something like: "begin_access is defined to have side effects, but this is not relevant for DCE."
Because this is the reason why you have to handle begin_access here.

Reply:
Fair enough. Done.